app/src/main/kotlin/com/intive/picover/images/viewmodel/ImagesViewModel.kt

```diff
@@ -0,0 +1,47 @@
+package com.intive.picover.images.viewmodel
+
+import android.net.Uri
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.intive.picover.images.repository.ImagesRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+class ImagesViewModel @Inject constructor(private val repository: ImagesRepository) : ViewModel() {
+
+	private val _uiState = MutableStateFlow<LatestFirebaseStorageState>(LatestFirebaseStorageState.Loading)
+	val uiState: StateFlow<LatestFirebaseStorageState> = _uiState
+
+	init {
+		viewModelScope.launch(Dispatchers.IO) {
```

Comment:
moving execution to safe thread should be done in a place where you do non-ui operations.

Reply:
Good point. Thank you.